### ISSUE TITLE
A false negative about  the rule BC_IMPOSSIBLE_DOWNCAST

### ISSUE THREAD

**Issue by RJerrica** [spotbugs/spotbugs#3769]
Hi, I found a false negative regarding the rule **BC_IMPOSSIBLE_DOWNCAST**. In the following code example, SpotBugs should have reported a warning of this type at line 7, but no warnings were actually reported. However, since `HashMap` is incorrectly cast to `LinkedHashMap`, this is a false negative bug.

### Code Example
```java
import java.util.HashMap;
import java.util.LinkedHashMap;

public class Main {
    public static void main(String args[]) {
        HashMap map = new HashMap();
        LinkedHashMap linkedHashMap = (LinkedHashMap) map.clone();
    }
}
```

SpotBugs version: lastest
